### ISSUE TITLE
`AttributeError: 'Version' object has no attribute 'release'` after upgrade to 6.1.0

### ISSUE THREAD

**Issue by P403n1x87** [pypa/setuptools-scm#605]
We have jobs in CI that depend on the latest version of this extension and so as soon as 6.1.0 became available on PyPI we started hitting the following error:

~~~
Traceback (most recent call last):
  File "setup.py", line 150, in <module>
    setup(
  File "/usr/lib/python3/dist-packages/setuptools/__init__.py", line 144, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.8/distutils/core.py", line 108, in setup
    _setup_distribution = dist = klass(attrs)
  File "/usr/lib/python3/dist-packages/setuptools/dist.py", line 447, in __init__
    _Distribution.__init__(self, {
  File "/usr/lib/python3.8/distutils/dist.py", line 292, in __init__
    self.finalize_options()
  File "/usr/lib/python3/dist-packages/setuptools/dist.py", line 740, in finalize_options
    ep.load()(self)
  File "/usr/lib/python3/dist-packages/setuptools/dist.py", line 747, in _finalize_setup_keywords
    ep.load()(self, ep.name, value)
  File "/home/circleci/project/.eggs/setuptools_scm-6.1.0-py3.8.egg/setuptools_scm/integration.py", line 26, in version_keyword
    dist.metadata.version = _get_version(config)
  File "/home/circleci/project/.eggs/setuptools_scm-6.1.0-py3.8.egg/setuptools_scm/__init__.py", line 188, in _get_version
    dump_version(
  File "/home/circleci/project/.eggs/setuptools_scm-6.1.0-py3.8.egg/setuptools_scm/__init__.py", line 94, in dump_version
    version_fields = parsed_version.release
AttributeError: 'Version' object has no attribute 'release'
~~~

Our `setup.py` has the following

~~~
    use_scm_version={"write_to": "ddtrace/_version.py"},
    setup_requires=["setuptools_scm[toml]>=4", "cython"],
~~~

and we also have

~~~
"packaging>=17.1",
~~~
in `install_requires`.

**Comment by RonnyPfannschmidt** [comment]
I'll get to debugging this as soon as the toddler is sleeping

**Comment by crockeo** [comment]
I'll leave it up to maintainers to provide a definitive answer, but it looks like this occurs because of an incompatibility between an old version of `setuptools` and a new version of `setuptools_scm`. [This commit](https://github.com/pypa/setuptools/commit/bf069fe9ddcadaa2c029067601d06a07d037d4f7#diff-029a51295b7f119c0d91c6a5c0e5a7b4945640cc0848b7bd5e64bb05c9d88cc9R280-R282) from October 2019 introduced the `release` field to `Version`s in the vendored copy of [packaging](https://pypi.org/project/packaging/) bundled with `setuptools` which is included in any release after `setuptools==42.0.0`.

Tl;dr upgrade setuptools to `setuptools>=42.0.0`.

**Comment by RonnyPfannschmidt** [comment]
I suspect its a missed compatibility test between setuptools and packaging, please try a setuptools update as well (i believe i missed that old setuptools vendors a too old packaging)

Any setuptools that tries to use pip instead of easy install is likely recent enough

**Comment by RonnyPfannschmidt** [comment]
@crockeo thanks for verifying my suspicion faster than I send it out

**Comment by crockeo** [comment]
Just dealt with a support ticket about this earlier today, so I wanted to share my findings upstream 🙂

**Comment by RonnyPfannschmidt** [comment]
yanked the release now that i'm back at the computer, will work on a fix now

**Comment by RonnyPfannschmidt** [comment]
i finished analysis - setuptools is ignoring the requirement for `setuptools >= 45`

the  line in https://github.com/pypa/setuptools_scm/blob/main/setup.py#L31 should have sorted the issue, it didn't

**Comment by henryiii** [comment]
setup.py isn't run (or included) when you install from a wheel, so virtually no one should see that error.

**Comment by RonnyPfannschmidt** [comment]
@henryiii the error happens in a build egg location, so it is executed

**Comment by mspang** [comment]
I'd suggest finding a way to retain compatibility with older setuptools as virtualenvs with older setuptools than you require are produced by default on every system I've tried.

**Comment by henryiii** [comment]
(Not that I like being stuck with old versions of things, but I agree - upgrading setuptools as a requirement has always been very hard to do. It's solved by PEP 517, but not everyone is using that (though I'm not sure why not).)

**Comment by muthu-tech** [comment]
Pip installation that was working with python36-devel stopped working while called via DockerFile. And I tried setuptools latest version, still it throws  AttributeError: 'Version' object has no attribute 'release'. 

Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-tzjqsevs/**petl**/

Can someone please let me know any options worth trying to get back working pip installations via docker

**Comment by RonnyPfannschmidt** [comment]
@muthu-tech update pip and setuptools

**Comment by muthu-tech** [comment]
> @muthu-tech update pip and setuptools

@RonnyPfannschmidt  I did that . I have ,
`python3 -m pip install --upgrade pip`

`python3 -m pip  install -r requirements.txt`

and first entry in requirements.txt has setuptools==57.4.0

And it throws 

ERROR: Could not install packages due to an OSError: [Errno 39] Directory not empty: '__pycache__'


FYI, these commands are inside a dockerfile and pip installations were working fine until last week and right now they are broken.

**Comment by RonnyPfannschmidt** [comment]
Update setuptools first in requirements is too late
